src/odin/ingestion/qlik/cubic_archive.py

```diff
@@ -0,0 +1,354 @@
+            if self.save_local:
+                move_path = str(new_path).replace(self.tmpdir, ".")
+                os.makedirs(os.path.dirname(move_path), exist_ok=True)
+                shutil.move(new_path, move_path)
+            # Upload merged parquet paths to S3
+            else:
+                move_path = str(new_path).lstrip(self.tmpdir)
+                upload_file(new_path, move_path)
+
+    def process_snapshot_group(self, group: List[Tuple[str, QlikDFM]]) -> None:
+        """
+        Convert group of snapshot files to parquet and sync to S3.
+
+        All snapshot files MUST be processed. Any error encountered during shapshot loading will
+        abort the process.
+
+        Process:
+            1. Download each snapshot CSV file
+            2. Convert each CSV to parquet
+            3. Sync all parquet files to S3
+        """
+        if len(group) == 0:
+            return
+        log = ProcessLog("load_qlik_snapshot_group", group_len=len(group))
+        assert group[0][0].endswith("0001.csv.gz")
+        snap_dt = dfm_snapshot_dt(group[0][1])
+        snap_part = f"snapshot={snap_dt.strftime(SNAPSHOT_FMT)}"
+        part_folder = os.path.join(self.tmpdir, snap_part)
+        os.makedirs(part_folder, exist_ok=True)
+        pq_snap_paths = []
+        for path, dfm in group:
+            pq_snap_paths.append(snapshot_to_parquet(path, dfm, part_folder))
+            self.archive_objects.append(path)
+
+            # If more than 20 snapshot files are being processed, they will be "synced"
+            # in batches of 20. This limits to amount of local storage used during the
+            # snapshot load process.
+            if len(pq_snap_paths) > 19:
+                self.sync_tmp_paths(pq_snap_paths)
+                self.reset_tmpdir()
+                part_folder = os.path.join(self.tmpdir, snap_part)
+                os.makedirs(part_folder, exist_ok=True)
+                pq_snap_paths = []
+
+        self.sync_tmp_paths(pq_snap_paths)
+        self.reset_tmpdir()
+        log.complete()
+
+    def load_snapshots(self) -> None:
+        """
+        Archive Qlik LOAD file(s) to parquet partitions.
+
+        In the case that there are multiple Qlik snapshots are available for processsing, this
+        function will group files belonging to the same snapshot together.
+
+        All of the LOAD files, belonging to a snapshot group, are then processed individually and
+        added to the same snapshot partition.
+        """
+        snapshot_group: List[Tuple[str, QlikDFM]] = []
+        for path, dfm in find_qlik_load_files(self.table):
+            if path.endswith("00001.csv.gz"):
+                self.process_snapshot_group(snapshot_group)
+                snapshot_group.clear()
+            snapshot_group.append((path, dfm))
+        self.process_snapshot_group(snapshot_group)
+
+    def group_changes(self, change_files: List[S3Object]) -> Dict[str, List[S3Object]]:
```

Comment:
'change' -> 'cdc'. See other comment about consistency and it being more informative.

Reply:
Incorporated!